libfdt/meson.build

```diff
@@ -26,27 +26,33 @@ else
 endif
 
 link_args += version_script
-libfdt = both_libraries(
+libfdt = library(
   'fdt', sources,
   version: meson.project_version(),
   link_args: link_args,
   link_depends: 'version.lds',
   install: true,
 )
 
-if static_build
-  link_with = libfdt.get_static_lib()
-else
-  link_with = libfdt.get_shared_lib()
-endif
-
 libfdt_inc = include_directories('.')
 
 libfdt_dep = declare_dependency(
   include_directories: libfdt_inc,
-  link_with: link_with,
+  link_with: libfdt,
 )
 
+if static_build
+  libfdt_static_dep = declare_dependency(
+    include_directories: libfdt_inc,
```

Comment:
Shouldn't the include directories me listed in both static and dynamic cases?

Reply:
In the dynamic case it uses the `libfdt_dep` which is itself a dependency that has already included `libfdt_inc`.